internal/service/alertconfiguration/model_alert_configuration.go

```diff
@@ -200,28 +200,28 @@ func NewTFMetricThresholdConfigModel(t *admin.ServerlessMetricThreshold, currSta
 		return []TfMetricThresholdConfigModel{
 			{
 				MetricName: conversion.StringNullIfEmpty(t.MetricName),
-				Operator:   conversion.StringNullIfEmpty(*t.Operator),
-				Threshold:  types.Float64Value(*t.Threshold),
-				Units:      conversion.StringNullIfEmpty(*t.Units),
-				Mode:       conversion.StringNullIfEmpty(*t.Mode),
+				Operator:   conversion.StringNullIfEmpty(t.GetOperator()),
+				Threshold:  types.Float64Value(t.GetThreshold()),
+				Units:      conversion.StringNullIfEmpty(t.GetUnits()),
+				Mode:       conversion.StringNullIfEmpty(t.GetMode()),
```

Comment:
just to fully understand, how where you able to have the API respond with a MetricThreshold object that only had some fields defined? Which one was not being returned?

Reply:
The issue was for `threshold_config` but I also did changes on `metric_threshold_config` just to be safe. 
The issue can be reproduced following this steps (more details [here](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/2458#issuecomment-2252309821)):
- Create an alert_notification without threshold_config 
- Through the Atlas UI, edit the alert configuration created in terraform and set a threshold_config
- terraform apply
- nil pointer dereference happens and provider crashes

Reply:
in these cases it will great to create a test that fails before the fix and passes after the fix is applied

Reply:
I have not been able to create a test for this because to reproduce this issue the resource has to be modified from outside TF. I don't think we currently are able to reproduce this cases in our tests. Open to suggestions though :)